### ISSUE TITLE
when mulitple auth types are used, it is handy to be able to set wantsurl via login url

### ISSUE THREAD

**Issue by danmarsden** [catalyst/moodle-auth_saml2#191]
eg login url like this:
...auth/saml2/login.php?wantsurl=http://../course/view.php?id=6

but because login.php doesn't check wantsurl before redirecting to external login it isn't saved and the user is just redirected to the homepage.

**Comment by garthwilliamson** [comment]
If it's suitable for you, we're using the "wants" parameter that this plugin adds to implement this. I can make a proper patch if this is suitable:

commit f3b36afecdc6daec888cb96b5b7ce58a256fb244 (HEAD, refs/remotes/origin/unicef-WR287403)
Author: Garth Williamson <garth@catalyst-eu.net>
Date:   Tue Mar 6 11:37:54 2018 +0000

    WR287403: set wantsurl into session

diff --git a/auth.php b/auth.php
index f331db2..bd52d76 100644
--- a/auth.php
+++ b/auth.php
@@ -356,6 +356,13 @@ class auth_plugin_saml2 extends auth_plugin_base {
         }
 
         $auth = new SimpleSAML_Auth_Simple($this->spname);
+
+        // CATALYST CUSTOM.
+        if (isset($_GET['wants'])) {
+            $SESSION->wantsurl = $_GET['wants'];
+        }
+        // CATALYST CUSTOM END.
+
         $auth->requireAuth();
         $attributes = $auth->getAttributes();

**Comment by garthwilliamson** [comment]
And I have one further change already:         if (isset($_GET['wants']) && !isset($SESSION->wantsurl)) {

to deal with users who get mutliple redirects.

**Comment by danmarsden** [comment]
yeah - something like that that doesn't use raw $_GET (bad Garth!) :-)

usually something like this is used:
$wantsurl = new moodle_url(optional_param('wantsurl', '', PARAM_URL));

except in the case of mnet when the url may not be for the current site.
